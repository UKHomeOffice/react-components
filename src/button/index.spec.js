import React from 'react'
import { render } from 'enzyme'
import Button from 'govuk-react-components/button';

describe('The Button Component', () => {

  it('should render children that are passed in', () => {
    expect(
      render(<Button>Sample Text</Button>)
    ).toMatchSnapshot()
  });

  it('should append classNames when attribute is passed in', () => {
    expect(
      render(<Button className="my-additional-classname">Sample Text</Button>)
    ).toMatchSnapshot();
  });

  it('should disable when disabled attribute is passed in', () => {
    expect(
      render(<Button disabled={true}>Sample Text</Button>)
    ).toMatchSnapshot();
  });

});
