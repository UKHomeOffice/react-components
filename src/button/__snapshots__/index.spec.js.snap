// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Standard Button Component should append classNames when attribute is passed in 1`] = `
<button
  class="button my-additional-classname"
>
  Sample Text
</button>
`;

exports[`The Standard Button Component should disable when disabled attribute is passed in 1`] = `
<button
  class="button"
  disabled=""
>
  Sample Text
</button>
`;

exports[`The Standard Button Component should render children that are passed in 1`] = `
<button
  class="button"
>
  Sample Text
</button>
`;

exports[`The link Button Component should append classNames when attribute is passed in 1`] = `
<a
  class="button my-additional-classname"
  href="#"
>
  Sample Text
</a>
`;

exports[`The link Button Component should disable when disabled attribute is passed in 1`] = `
<a
  class="button"
  disabled=""
  href="#"
>
  Sample Text
</a>
`;

exports[`The link Button Component should render as an <a> tag with the mandatory href 1`] = `
<a
  class="button"
  href="#test-href"
>
  Sample Text
</a>
`;
